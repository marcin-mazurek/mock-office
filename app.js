const express = require('express');
require('./dist/double-rest-api');

const serve = express.static('./dist/double-gui');
const app = express();
const server = app.use(serve);

server.listen(3070, () => {
  console.log('GUI address: http://127.0.0.1:3070');
});
