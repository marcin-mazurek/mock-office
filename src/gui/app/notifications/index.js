export * from './selectors';
export * from './createNotification';
export { default as reducer } from './reducer';
export * from './actions';
