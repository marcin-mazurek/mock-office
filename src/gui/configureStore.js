import { fromJS } from 'immutable';
import { createStore, compose, applyMiddleware } from 'redux';
import { hashHistory } from 'react-router';
import { routerMiddleware as createRouterMiddleware } from 'react-router-redux';
import { createEpicMiddleware } from 'redux-observable';
import 'rxjs';
import rootReducer from './app/rootReducer';
import rootEpic from './store/rootEpic';

const epicMiddleware = createEpicMiddleware(rootEpic);

const DEBUG = process.env.NODE_ENV !== 'production';

export default () => {
  const initialState = fromJS({});
  const routerMiddleware = createRouterMiddleware(hashHistory);
  const enhancer = compose(
    applyMiddleware(routerMiddleware, epicMiddleware),
    (DEBUG && window.devToolsExtension) ? window.devToolsExtension() : f => f
  );
  return createStore(rootReducer, initialState, enhancer);
};
