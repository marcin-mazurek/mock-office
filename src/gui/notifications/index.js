import {
  SUCCEEDED as ADD_SERVER_DID_SUCCEED,
  FAILED as ADD_SERVER_FAILED
} from '../servers/addServer/epics';
import { FAILED as EDIT_SERVER_DID_FAIL } from '../servers/editServer/epics';
import {
  FAILED as REMOVE_SERVER_DID_FAIL,
  SUCCEEDED as REMOVE_SERVER_DID_SUCCEED
} from '../servers/removeServer/epics';
import { DID_FAIL as REMOVE_MOCK_DID_FAIL } from '../mocks/removeMock/epics';
import {
  SUCCEEDED as IMPORT_MOCKS_SUCCEEDED,
  FAILED as IMPORT_MOCKS_FAILED
} from '../mocks/importMock/epics';
import {
  SUCCEED as ADD_HTTP_MOCK_SUCCEED,
  FAILED as ADD_HTTP_MOCK_FAILED
} from '../modules/addHttpMock/epic';
import {
  SUCCEED as ADD_WS_MOCK_SUCCEED,
  FAILED as ADD_WS_MOCK_FAILED
} from '../modules/addWsMock/epic';
import {
  FAILED as START_SERVER_FAILED
} from '../servers/startServer/startServer';
import { NOTIFICATION_CLICKED } from './NotificationsList';
import notificationsReduxModule from './notificationsReduxModule';
import createNotificationsListConnect from './createNotificationsListConnect';
import { FAILED as STOP_SERVER_FAILED } from '../servers/stopServer';
import { NOTIFICATION_EXPIRED } from './clearExpiredNotificationsEpic';

const showError = (state, action, reducers) =>
  reducers.addNotification(state, { type: 'error', text: action.reason });
const showAddMockSuccess = (state, action, reducers) =>
  reducers.addNotification(state, { type: 'success', text: 'Mock added' });

const notifications = notificationsReduxModule(
  {
    [ADD_SERVER_DID_SUCCEED]: (state, action, reducers) =>
      reducers.addNotification(state, { type: 'success', text: 'Server added' }),
    [REMOVE_SERVER_DID_SUCCEED]: (state, action, reducers) =>
      reducers.addNotification(state, { type: 'success', text: 'Server removed' }),
    [EDIT_SERVER_DID_FAIL]: showError,
    [REMOVE_SERVER_DID_FAIL]: showError,
    [REMOVE_MOCK_DID_FAIL]: showError,
    [IMPORT_MOCKS_FAILED]: showError,
    [ADD_HTTP_MOCK_FAILED]: showError,
    [ADD_WS_MOCK_FAILED]: showError,
    [ADD_SERVER_FAILED]: showError,
    [START_SERVER_FAILED]: showError,
    [STOP_SERVER_FAILED]: showError,
    [IMPORT_MOCKS_SUCCEEDED]: (state, action, reducers) =>
      reducers.addNotification(state, { type: 'success', text: 'Mocks imported' }),
    [ADD_HTTP_MOCK_SUCCEED]: showAddMockSuccess,
    [ADD_WS_MOCK_SUCCEED]: showAddMockSuccess,
    [NOTIFICATION_CLICKED]: (state, action, reducers) =>
      reducers.removeNotification(state, action.id),
    [NOTIFICATION_EXPIRED]: (state, action, reducers) =>
      reducers.removeNotification(state, action.id)
  },
  state => state.get('notifications')
);

export const reducer = notifications.reducer;
export const selectors = notifications.selectors;
export const NotificationsListConnect = createNotificationsListConnect(notifications.selectors);
