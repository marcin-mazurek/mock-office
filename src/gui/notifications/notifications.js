import { SUCCEED as ADD_SERVER_DID_SUCCEED, FAILED as ADD_SERVER_FAILED } from '../servers/addServer/epics';
import { DID_FAIL as EDIT_SERVER_DID_FAIL } from '../servers/editServer/epics';
import {
  DID_FAIL as REMOVE_SERVER_DID_FAIL,
  DID_SUCCEED as REMOVE_SERVER_DID_SUCCEED
} from '../servers/removeServer/epics';
import { DID_FAIL as REMOVE_MOCK_DID_FAIL } from '../mocks/removeMock/epics';
import {
  SUCCEEDED as IMPORT_MOCKS_SUCCEEDED,
  FAILED as IMPORT_MOCKS_FAILED
} from '../mocks/importMock/epics';
import {
  SUCCEED as ADD_HTTP_MOCK_SUCCEED,
  FAILED as ADD_HTTP_MOCK_FAILED
} from '../mocks/addMock/addHttpMock/epic';
import {
  SUCCEED as ADD_WS_MOCK_SUCCEED,
  FAILED as ADD_WS_MOCK_FAILED
} from '../mocks/addMock/addWsMock/epic';
import {
  FAILED as START_SERVER_FAILED
} from '../servers/startServer/epics';
import notificationsReduxModule from './notificationsReduxModule';

const showError = (state, action, module) =>
  module.addNotification(state, { type: 'error', text: action.reason });
const showAddMockSuccess = (state, action, module) =>
  module.addNotification(state, { type: 'success', text: 'Mock added' });

export const reducer = notificationsReduxModule.createReducer({
  [ADD_SERVER_DID_SUCCEED]: (state, action, module) =>
    module.addNotification(state, { type: 'success', text: 'Server added' }),
  [REMOVE_SERVER_DID_SUCCEED]: (state, action, module) =>
    module.addNotification(state, { type: 'success', text: 'Server removed' }),
  [EDIT_SERVER_DID_FAIL]: showError,
  [REMOVE_SERVER_DID_FAIL]: showError,
  [REMOVE_MOCK_DID_FAIL]: showError,
  [IMPORT_MOCKS_FAILED]: showError,
  [ADD_HTTP_MOCK_FAILED]: showError,
  [ADD_WS_MOCK_FAILED]: showError,
  [ADD_SERVER_FAILED]: showError,
  [START_SERVER_FAILED]: showError,
  [IMPORT_MOCKS_SUCCEEDED]: (state, action, module) =>
    module.addNotification(state, { type: 'success', text: 'Mocks imported' }),
  [ADD_HTTP_MOCK_SUCCEED]: showAddMockSuccess,
  [ADD_WS_MOCK_SUCCEED]: showAddMockSuccess
});
export const selectors = notificationsReduxModule.createSelectors(state => state.get('notifications'));
