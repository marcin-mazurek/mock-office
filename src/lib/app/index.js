export * from './appServer';
export { default as configurePersistentState } from './createPersistentState';
export * from './guiEventsServer';
export { serversManager } from './serversManager';
export * from './guiServer';
