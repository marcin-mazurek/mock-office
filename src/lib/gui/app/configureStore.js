import { fromJS } from 'immutable';
import { createStore, compose, applyMiddleware } from 'redux';
import { hashHistory } from 'react-router';
import { routerMiddleware as createRouterMiddleware } from 'react-router-redux';
import { createEpicMiddleware } from 'redux-observable';
import { reducer } from './index';
import { epic } from '../epics/index';

const epicMiddleware = createEpicMiddleware(epic);

const DEBUG = process.env.NODE_ENV !== 'production';

export default () => {
  const initialState = fromJS({});
  const routerMiddleware = createRouterMiddleware(hashHistory);
  const enhancer = compose(
    applyMiddleware(routerMiddleware, epicMiddleware),
    (DEBUG && window.devToolsExtension) ? window.devToolsExtension() : f => f
  );
  return createStore(reducer, initialState, enhancer);
};
