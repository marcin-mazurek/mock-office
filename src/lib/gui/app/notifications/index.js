import { createReducer } from 'notifications';
import {
  SUCCEEDED as ADD_SERVER_SUCCEEDED,
  FAILED as ADD_SERVER_FAILED
} from '../../epics/addServer/actions';
import { FAILED as EDIT_SERVER_FAILED } from '../../epics/editServer/actions';
import {
  FAILED as REMOVE_SERVER_FAILED,
  SUCCEEDED as REMOVE_SERVER_SUCCEEDED
} from '../../epics/removeServer/actions';
import { DID_FAIL as REMOVE_BEHAVIOUR_DID_FAIL } from '../../epics/removeBehaviour';
import {
  SUCCEEDED as IMPORT_BEHAVIOURS_SUCCEEDED,
  FAILED as IMPORT_BEHAVIOURS_FAILED
} from '../../epics/importBehaviour';
import {
  SUCCEED as ADD_BEHAVIOUR_SUCCEED,
  FAILED as ADD_BEHAVIOUR_FAILED
} from '../../epics/addBehaviour';
import {
  FAILED as START_SERVER_FAILED
} from '../../epics/startServer/actions';
import { FAILED as STOP_SERVER_FAILED } from '../../epics/stopServer';
import { SUBMIT_FAILED } from '../../components/AddHttpBehaviourForm/actions';

export default createReducer(
  {
    [ADD_BEHAVIOUR_SUCCEED]: reduce => reduce('addNotification', {
      type: 'success',
      text: 'Behaviour added'
    }),
    [IMPORT_BEHAVIOURS_SUCCEEDED]: reduce => reduce('addNotification', {
      type: 'success',
      text: 'Behaviours imported'
    }),
    [EDIT_SERVER_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [REMOVE_SERVER_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [REMOVE_BEHAVIOUR_DID_FAIL]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [IMPORT_BEHAVIOURS_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [ADD_BEHAVIOUR_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [ADD_SERVER_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [START_SERVER_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [STOP_SERVER_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [SUBMIT_FAILED]: (reduce, action) => reduce('addNotification', {
      type: 'error',
      text: action.reason
    }),
    [REMOVE_SERVER_SUCCEEDED]: reduce => reduce('addNotification', {
      type: 'success',
      text: 'Server removed'
    }),
    [ADD_SERVER_SUCCEEDED]: reduce => reduce('addNotification', {
      type: 'success',
      text: 'Server added'
    })
  },
  true
);
