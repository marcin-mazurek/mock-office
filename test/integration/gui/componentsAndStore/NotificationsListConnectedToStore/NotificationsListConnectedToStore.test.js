import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from '../../../../../src/lib/gui/app/configureStore';
import {
  NotificationsListConnect,
  NotificationsList,
  NotificationListItem
} from '../../../../../src/lib/gui/components/NotificationsList';
import { addNotificationAction } from '../../../../../src/lib/gui/app/notifications';

describe('NotificationsList', () => {
  test('should display notifications from redux store', () => {
    const store = configureStore();
    const wrapper = mount(
      <Provider store={store}>
        <NotificationsListConnect />
      </Provider>
    );

    expect(wrapper.find(NotificationsList).exists()).toBeTruthy();
    store.dispatch(
      addNotificationAction({
        text: 'message'
      })
    );
    expect(wrapper.find(NotificationListItem)).toHaveLength(1);
  });
});
