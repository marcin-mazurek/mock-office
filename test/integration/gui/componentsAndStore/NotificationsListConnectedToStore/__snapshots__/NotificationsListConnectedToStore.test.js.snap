// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsList should display notifications from redux store 1`] = `
<NotificationsList
  notifications={
    Array [
      Object {
        "id": "cj71zdore0000fzve5gmwovyh",
        "text": "message",
        "type": "info",
      },
    ]
  }
  onNewNotificationDisplayed={[Function]}
  onNotificationClick={[Function]}
>
  <ul
    className="notifications-list"
  >
    <li
      className="notifications-list-item"
    >
      <button
        className="notifications-bubble notifications-bubble--info"
        onClick={[Function]}
      >
        message
      </button>
    </li>
  </ul>
</NotificationsList>
`;
